Framework/DataHandling/inc/MantidDataHandling/LoadEventNexus.h

```diff
@@ -636,34 +636,36 @@ void LoadEventNexus::loadEntryMetadata(const std::string &nexusfilename, T WS,
 
   // get the sample name - nested try/catch to leave the handle in an
   // appropriate state
-  try {
+  if (exists(file, "sample")) {
     file.openGroup("sample", "NXsample");
-    if (exists(file, "name")) {
-      file.openData("name");
-      const auto info = file.getInfo();
-      std::string name;
-      if (info.type == ::NeXus::CHAR) {
-        if (info.dims.size() == 1) {
-          name = file.getStrData();
-        } else { // something special for 2-d array
-          const int64_t total_length = std::accumulate(
-              info.dims.begin(), info.dims.end(), static_cast<int64_t>(1),
-              std::multiplies<int64_t>());
-          boost::scoped_array<char> val_array(new char[total_length]);
-          file.getData(val_array.get());
-          file.closeData();
-          name = std::string(val_array.get(), total_length);
+    try {
+      if (exists(file, "name")) {
+        file.openData("name");
+        const auto info = file.getInfo();
+        std::string name;
+        if (info.type == ::NeXus::CHAR) {
+          if (info.dims.size() == 1) {
```

Comment:
I suspect this check could be the real cause for the failure. This is a 2D in the file, but is a Fixed Length String.